docs/advanced_examples/KNearestNeighbors.ipynb

```diff
@@ -0,0 +1,431 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# K-Nearest Neighbors\n",
+    "\n",
+    "\n",
+    "K-Nearest Neighbors (KNN) is a non-parametric supervised learning algorithm, suitable for both classification and regression tasks.\n",
+    "\n",
+    "In classification, KNN aims to identify the nearest points by measuring their similarity, often through distance metrics. The new labels are then assigned through majority voting, considering the most frequent labels among the neighboring points.\n",
+    "\n",
+    "In Fully Homomorphic Encryption (FHE), classification with KNN poses significant computational challenges due to the distance calculations and the sorting algorithms, which is currently a non-stable algorithm.\n",
```

Comment:
can you clarify what we mean by `which is currently a non-stable algorithm` ?

Reply:
added a small sentence.